mapiproxy/servers/default/emsmdb/oxcfold.c

```diff
@@ -200,7 +201,15 @@ _PUBLIC_ enum MAPISTATUS EcDoRpc_RopGetHierarchyTable(TALLOC_CTX *mem_ctx,
 		mapi_repl->error_code = MAPI_E_INVALID_OBJECT;
 		goto end;
 	}
+
+	object->object.table->flags = mapi_req->u.mapi_GetHierarchyTable.TableFlags;
+
 	mapi_handles_set_private_data(rec, object);
+
+	if (object->object.table->flags & TableFlags_Depth) {
+		emsmdbp_folder_get_recursive_folder_count(emsmdbp_ctx, parent_object, &count);
```

Comment:
We aren't checking the `retval`, so the `count` value can be incorrect or corrupted, can't it?